### DISCUSSION TITLE
How to access resolver from another resolver in Apollo Graphql

### DISCUSSION THREAD

**Discussion by GonzaloSP** [apollographql/apollo-server#6598] [Q&A]
I have two resolvers in ApolloGraphQL. I want to access the Query defined by ResolverB in ResolverA. What's the best way to do it?

```
const resolvers = {
    ResolverA: {
        quote: (root, args, context) => {
            return getQuote()
        }
    },
    ResolverB: {
         resolverbQuery: (root, args, context) => {                    
            return ResolverA.quote()
         }
    }
const server = new ApolloServer({typeDefs, resolvers, context: { getQuote, logger }})

const app = express()

server.applyMiddleware( {app} )

app.listen({port: 4000}, () => {
    console.log('Server is running')
})
```

**Comment by glasser** [comment]
You can actually just call it directly: `resolvers.ResolverA.quote`, since `resolvers` is just a plain old JS object.  But you'd need to pass appropriate values for all the arguments, and perhaps their root/parent arguments expect different types. It probably is simpler to just have both resolvers call a common function.

**Reply by GonzaloSP** [reply]
@glasser thanks for the reply. 

The issue is that ResolverB doesn't receive resolvers as one of the parameters. How can I import it?

**Reply by glasser** [reply]
In the code you provided, `resolvers` is defined in the scope you've given, so it should just work.
